packages/data/src/types.d.ts

```diff
@@ -0,0 +1,36 @@
+export type WPDataFunctionOrGeneratorArray = {
+    [index: number]: Function|Generator;
+};
+export type WPDataFunctionArray = {
+    [index: number]: Function;
+};
```

Comment:
I think these can be simplified:

```suggestion
export type WPDataFunctionOrGeneratorArray = Array< Function | Generator >;
export type WPDataFunctionArray = Array< Function >;
```